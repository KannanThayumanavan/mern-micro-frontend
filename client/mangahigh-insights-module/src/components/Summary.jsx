import React from 'mangahigh-libs/react';
import DisplayCard from 'mangahigh-components/DisplayCard';
import PropTypes from 'prop-types';

const calculateAverage = (attempts) => attempts
	.map((attempt) => attempt.score)
	.reduce((accumulator, currentValue) => accumulator + currentValue) / attempts.length;

const Summary = ({ attempts }) =>  (	
	<div className="row">			
		<div className="col-md-6">
			<DisplayCard
		    highlight={(attempts.length).toString()}
		    summary='Number of times played'
		  />
	  </div>	
	  <div className="col-md-6">
		  <DisplayCard
		    highlight={(calculateAverage(attempts).toFixed(1)).toString()}
		    summary='Average score'
		  />
		</div>		
  </div>
);

Summary.propTypes = {
	attempts: PropTypes.arrayOf(PropTypes.shape({
		date_and_time: PropTypes.string,
		score: PropTypes.number,
		total_questions: PropTypes.number,
	})).isRequired
};

export default Summary;