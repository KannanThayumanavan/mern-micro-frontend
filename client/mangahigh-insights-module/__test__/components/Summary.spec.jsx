import React from 'react';
import {shallow } from 'enzyme';
import Summary from '../../src/components/Summary';

jest.mock('mangahigh-libs/react', () => jest.requireActual('react'),
	{ virtual: true }
);

jest.mock('mangahigh-components/DisplayCard', () => () => 'DisplayCard'			
, { virtual: true });

describe('Test Summary Component', () => {
	const attempts = [
		{
			"_id":"6161b01c74f86b0917b418c8",
			"date_and_time":"2021-10-09T15:07:07.763Z",
			"score":3,
			"total_questions":3
		},
		{
			"_id":"6161b01c74f86b0917b418b6",
			"date_and_time":"2021-10-09T15:37:07.763Z",
			"score":1,
			"total_questions":3
		}
	];
	it('should render correctly with required props', () => {
		const renderedModule = shallow(<Summary attempts={attempts} />);
    expect(renderedModule).toMatchSnapshot();
	});
});