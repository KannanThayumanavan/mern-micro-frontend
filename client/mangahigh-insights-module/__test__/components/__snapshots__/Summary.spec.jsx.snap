// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Summary Component should render correctly with required props 1`] = `
<div
  className="row"
>
  <div
    className="col-md-6"
  >
    <Component
      highlight="2"
      summary="Number of times played"
    />
  </div>
  <div
    className="col-md-6"
  >
    <Component
      highlight="2.0"
      summary="Average score"
    />
  </div>
</div>
`;
